src/rewards/veterans.rs

```diff
@@ -1,32 +1,322 @@
-use serde::Deserialize;
+use crate::community_advisors::models::{
+    AdvisorReviewId, AdvisorReviewRow,
+    ReviewRanking::{self, *},
+};
+use crate::rewards::Rewards;
+use itertools::Itertools;
+use rust_decimal::{prelude::ToPrimitive, Decimal};
+use std::borrow::Borrow;
+use std::collections::{BTreeMap, HashMap};
+
+use serde::{Deserialize, Serialize};
 
 pub type VeteranAdvisorId = String;
-pub type VeteranAdvisorReward = rust_decimal::Decimal;
 
-pub type VeteranAdvisorRewards = Vec<(VeteranAdvisorId, VeteranAdvisorReward)>;
+#[derive(Serialize)]
+pub struct VeteranAdvisorIncentive {
+    pub rewards: Rewards,
+    pub reputation: u64,
+}
+
+pub type EligibilityThresholds = std::ops::RangeInclusive<usize>;
 
+// TODO: for the sake of clarity, introduce a different naming between ca reviews and vca ranking
+
+// Supposing to have a file with all the rankings for each review
+// e.g. something like an expanded version of a AdvisorReviewRow
+// [proposal_id, advisor, ratings, ..(other fields from AdvisorReviewRow).., ranking (good/excellent/filtered out), vca]
 #[derive(Deserialize)]
-pub struct VeteranReviewsCount {
-    name: VeteranAdvisorId,
-    #[serde(alias = "No. of Reviews")]
-    number_of_reviews: usize,
+pub struct VeteranRankingRow {
+    #[serde(flatten)]
+    advisor_review_row: AdvisorReviewRow,
+    vca: VeteranAdvisorId,
+}
+
+impl VeteranRankingRow {
+    fn score(&self) -> ReviewRanking {
+        self.advisor_review_row.score()
+    }
+
+    fn review_id(&self) -> AdvisorReviewId {
+        self.advisor_review_row.id()
+    }
+}
+
+fn calc_final_ranking_per_review(rankings: &[impl Borrow<VeteranRankingRow>]) -> ReviewRanking {
+    let rankings_majority = rankings.len() / 2;
```

Comment:
What is the type of `rankings_majority`? Is it going to lead to any possible mistakes? I'm always a bit wary of divisions. Maybe employ decimals here as well?